template.tpl

```diff
@@ -1230,10 +1241,37 @@ const WRAPPER_NAMESPACE = '_amplitude';
 
 // Print a log message and set the tag to failed state
 const fail = msg => {
-  log(LOG_PREFIX + 'Error: ' + msg);
   return data.gtmOnFailure();
 };
 
+// Merge two Objects
+const mergeObject = (basedObject, overwriteObject) => {
+  if (!basedObject || Object.keys(basedObject).length == 0) return overwriteObject;
+  if (!basedObject || Object.keys(overwriteObject).length == 0) return basedObject;
+
+  // Clone
+  const newObject = JSON.parse(JSON.stringify(basedObject));
```

Comment:
Will JSON parse throw an exception if the input is not a valid json?

Reply:
<img width="508" alt="image" src="https://github.com/amplitude/amplitude-browser-sdk-gtm-template/assets/4400939/63959d5c-34ce-4b60-928c-1b0adfdd3772">

Reply:
Checked the `is valid object` before this. But it makes sense to move the checking in the mergeObject.

Reply:
Why does it need to be checked beforehand? 
`user_properties` is better to be removed before cloning the object. 

```
      const isValidPropertiesObject = data.eventPropertiesObject && isValidObject(data.eventPropertiesObject);
      let eventProperties = propertiesBaisc;
      if (isValidPropertiesObject) {
        const cleanedPropertiesObject = JSON.parse(JSON.stringify(data.eventPropertiesObject));
        // remove the user_properties
        Object.delete(cleanedPropertiesObject, 'user_properties');
        eventProperties = mergeObject(propertiesBaisc, cleanedPropertiesObject);
      }
```

Reply:
The JSON in GTM is different with the JSON in js. Here is the GTM doc. https://developers.google.com/tag-platform/tag-manager/templates/api
If the value cannot be parsed (e.g. malformed JSON), the function will return undefined.